synapse/storage/data_stores/main/registration.py

```diff
@@ -342,7 +343,7 @@ def is_real_user(self, user_id):
         )
         return res
 
-    @cachedInlineCallbacks()
+    @cached()
```

Comment:
I'm not sure I understand this change, did you want to make `is_support_user` async and then await the `runInteraction`?

Reply:
`runInteraction` returns a Deferred, so I'm just returning it directly instead of yielding it and then creating another Deferred by annotating the function with `@cachedInlineCallbacks`. In the end, the calling function gets a Deferred either way, but now we don't needlessly create one.

Reply:
Ah, OK. I was thinking that `@cached()` didn't work with `Deferreds` for some reason. 🤷

Reply:
Do you mean `Coroutine`s? Looks like it does:

https://github.com/matrix-org/synapse/blob/65902e08c3f4449de9baa4e6466f126585f688b3/synapse/storage/data_stores/main/state.py#L81-L82

Reply:
No, I mean `Deferreds`.

Reply:
Oh, it probably doesn't, which is why one would use `@cachedInlineCallbacks` instead?